compile.js

```diff
@@ -181,10 +189,52 @@ function compileFieldWrite(ctx, field, name) {
     }
 }
 
+function compileMapRead(readCode, name, numRepeated) {
+    return (numRepeated ? '' : 'var ') + 'entry = ' + readCode + '; obj.' + name + '[entry.key] = entry.value';
+}
+
 function compileRepeatedWrite(ctx, field, numRepeated) {
     return 'for (' + (numRepeated ? '' : 'var ') +
         'i = 0; i < obj.' + field.name + '.length; i++) ' +
-        compileFieldWrite(ctx, field, field.name + '[i]');
+        compileFieldWrite(ctx, field, 'obj.' + field.name + '[i]');
+}
+
+function compileMapWrite(ctx, field, numRepeated) {
+    var name = 'obj.' + field.name;
+
+    return 'for (' + (numRepeated ? '' : 'var ') +
+        'i in ' + name + ') if (Object.prototype.hasOwnProperty.call(' + name + ', i)) ' +
```

Comment:
Do we have to do this guard, or simply assume those are simple object literals? And why would an object not have a `hasOwnProperty`? To support `Object.create(null)` cases?

Reply:
Since this is the write code, you don't have control over the objects sent to it. In my opinion, it's better to default to safe code and this could be relaxed where the speed is necessary.

Yes, `Object.create(null)` won't have a prototype chain and therefore no `hasOwnProperty`. I don't feel strongly about that, it's a technique I picked up from the Airbnb style guide.